taiga_contrib_ldap_auth_ext/connector.py

```diff
@@ -91,58 +151,20 @@ def login(username: str, password: str) -> tuple:
-                       user=service_user, password=service_pass, authentication=service_auth)
-    except Exception as e:
-        error = "Error connecting to LDAP server: %s" % e
-        raise LDAPConnectionError({"error_message": error})
-
     # search for user-provided login
     search_filter = '(|(%s=%s)(%s=%s))' % (
         USERNAME_ATTRIBUTE, username, EMAIL_ATTRIBUTE, username)
     if SEARCH_FILTER_ADDITIONAL:
         search_filter = '(&%s%s)' % (search_filter, SEARCH_FILTER_ADDITIONAL)
-    try:
-        c.search(search_base=SEARCH_BASE,
-                 search_filter=search_filter,
-                 search_scope=SUBTREE,
-                 attributes=[USERNAME_ATTRIBUTE,
-                             EMAIL_ATTRIBUTE, FULL_NAME_ATTRIBUTE],
-                 paged_size=5)
-    except Exception as e:
-        error = "LDAP login incorrect: %s" % e
-        raise LDAPUserLoginError({"error_message": error})
-
-    # we are only interested in user objects in the response
-    c.response = [r for r in c.response if 'raw_attributes' in r and 'dn' in r]
-    # stop if no search results
-    if not c.response:
-        raise LDAPUserLoginError({"error_message": "LDAP login not found"})
-
-    # handle multiple matches
-    if len(c.response) > 1:
-        raise LDAPUserLoginError(
-            {"error_message": "LDAP login could not be determined."})
 
-    # handle missing mandatory attributes
-    raw_attributes = c.response[0].get('raw_attributes')
-    if raw_attributes.get(USERNAME_ATTRIBUTE) or raw_attributes.get(EMAIL_ATTRIBUTE) or raw_attributes.get(FULL_NAME_ATTRIBUTE):
-        raise LDAPUserLoginError({"error_message": "LDAP login is invalid."})
-
-    # attempt LDAP bind
-    username = raw_attributes.get(USERNAME_ATTRIBUTE)[0].decode('utf-8')
-    email = raw_attributes.get(EMAIL_ATTRIBUTE)[0].decode('utf-8')
-    full_name = raw_attributes.get(FULL_NAME_ATTRIBUTE)[0].decode('utf-8')
-    try:
-        dn = str(bytes(c.response[0].get('dn'), 'utf-8'), encoding='utf-8')
-        Connection(server, auto_bind=auto_bind, client_strategy=SYNC,
-                   check_names=True, authentication=SIMPLE,
-                   user=dn, password=password)
-    except Exception as e:
-        error = "LDAP bind failed: %s" % e
-        raise LDAPUserLoginError({"error_message": error})
-
-    # LDAP binding successful, but some values might have changed, or
-    # this is the user's first login, so return them
-    return (username, email, full_name)
+    for server in server_ldap_list:
+        try:
+            c = Connection(server, auto_bind=auto_bind, client_strategy=SYNC, check_names=True,
+                           user=service_user, password=service_pass, authentication=service_auth)
+            data = connect_to_ldap_server(server, c, auto_bind, search_filter, password)
+        except Exception as e:
+            print("Connection to LDAP server was failed", flush=True)
```

Comment:
@AminovE99 Correct message should be `Failed to authenticate against LDAP {0}` and replace `{0}` by the LDAP `server` info.